bin/hdfcoinc/pycbc_page_foreground

```diff
@@ -31,32 +59,63 @@ if args.output_file.endswith('.xml') or args.output_file.endswith('.xml.gz'):
         err_msg = "If creating xml files must provide the single detector "
         err_msg += "trigger lists with --single-detector-triggers."
         raise ValueError(err_msg)
-    
+
 fortrigs = hdf.ForegroundTriggers(args.trigger_file, args.bank_file,
-                       sngl_files=args.single_detector_triggers,
-                       n_loudest=args.num_coincs_to_write)
+                                  sngl_files=args.single_detector_triggers,
+                                  n_loudest=args.num_coincs_to_write)
 
 if args.output_file.endswith('.html'):
+    # If hierarchical removals were done and the user wants to look at a
+    # particular stage of hierarchical removals (h_num_rm), remove that
+    # number of loudest triggers by stat from the list of triggers.
+    if h_num_rm >= 0:
+        fortrigs.ignore_x_loudest(h_num_rm)
+
     ifar = fortrigs.get_coincfile_array('ifar')
-    ifar_exc = fortrigs.get_coincfile_array('ifar_exc') 
+    ifar_exc = fortrigs.get_coincfile_array('ifar_exc')
     fap = fortrigs.get_coincfile_array('fap')
     fap_exc = fortrigs.get_coincfile_array('fap_exc')
     stat = fortrigs.get_coincfile_array('stat')
     time1 = fortrigs.get_coincfile_array('time1')
     time2 = fortrigs.get_coincfile_array('time2')
-
     mass1 = fortrigs.get_bankfile_array('mass1')
     mass2 = fortrigs.get_bankfile_array('mass2')
     spin1z = fortrigs.get_bankfile_array('spin1z')
     spin2z = fortrigs.get_bankfile_array('spin2z')
 
+    # Write the correct false alarm rates and other significance values
+    # to the table after h_num_rm hierarchical removals
+    if h_num_rm >= 0:
+        # Regenerate the list of inclusive ifar and fap from foreground
+        # trigger id's from fortrigs.get_coincfile_array and stat file
+        trig_id1_table = fortrigs.get_coincfile_array('trigger_id1')
+
+        trig_id1_hrm = f['foreground_h%s/trigger_id1' % h_num_rm][:]
+        trig_id2_hrm = f['foreground_h%s/trigger_id2' % h_num_rm][:]
+
+        ifar_hrm = f['foreground_h%s/ifar' % h_num_rm][:]
+        fap_hrm = f['foreground_h%s/fap' % h_num_rm][:]
+
+        idx1_list = []
```

Comment:
Why do we need the next two sections? We've read in the ifar and fap above, and have the trigger id values to look up the rest? Shouldn't this be upfront, and combined with the previous section so we also don't need the call to remove some arbitrary number.